contracts/truefi/CurvePool.sol

```diff
@@ -33,22 +39,39 @@ contract CurvePool is TruePool {
 
     function exit(uint256 amount) external override {
         require(amount <= balanceOf(msg.sender), "CurvePool: Cannot withdraw amount bigger than available balance");
+        require(amount <= curvePool.token().balanceOf(address(this)), "CurvePool: Not enough cTokens in pool");
 
-        uint256 minTokenAmount = curvePool.calc_withdraw_one_coin(amount, TUSD_INDEX).mul(99).div(100);
+        uint256 minTokenAmount = curvePool.calc_withdraw_one_coin(amount, TUSD_INDEX).mul(97).div(100);
 
         uint256 balanceBefore = currencyToken().balanceOf(address(this));
-        curvePool.remove_liquidity_one_coin(amount, TUSD_INDEX, minTokenAmount);
+        curvePool.remove_liquidity_one_coin(amount, TUSD_INDEX, minTokenAmount, false);
         uint256 balanceAfter = currencyToken().balanceOf(address(this));
         require(currencyToken().transfer(msg.sender, balanceAfter.sub(balanceBefore)));
         _burn(msg.sender, amount);
     }
 
-    function borrow(uint256 amount) external override {
-        // TODO
+    function borrow(uint256 expectedAmount) external override {
+        require(msg.sender == lender, "CurvePool: Only lender can borrow");
+
+        uint256[N_TOKENS] memory amounts = [0, 0, 0, expectedAmount];
+        uint256 roughCurveTokenAmount = curvePool.curve().calc_token_amount(amounts, false).mul(105).div(100);
+        require(roughCurveTokenAmount <= curvePool.token().balanceOf(address(this)), "ERROR CODE 0");
+
+        uint256 balanceBefore = currencyToken().balanceOf(address(this));
+        curvePool.remove_liquidity_one_coin(roughCurveTokenAmount, TUSD_INDEX, 0, false);
+        uint256 balanceAfter = currencyToken().balanceOf(address(this));
+
+        require(balanceAfter.sub(balanceBefore) >= expectedAmount, "CurvePool: Not enough tokens withdrawn");
+        currencyToken().transfer(msg.sender, expectedAmount);
     }
 
     function repay(uint256 amount) external override {
-        // TODO
+        require(msg.sender == lender, "CurvePool: Only lender can repay");
+        require(currencyToken().transferFrom(msg.sender, address(this), amount));
+
+        uint256 amountToDeposit = amount.add(currencyToken().balanceOf(address(this)));
+        uint256[N_TOKENS] memory amounts = [0, 0, 0, amountToDeposit];
+        curvePool.add_liquidity(amounts, 0);
```

Comment:
Maybe let's extract these lines responsible for curve liquidity providing to separate, internal function?

Reply:
Also let's add an event emission.